nmpolicy/internal/capture/result.go

```diff
@@ -17,17 +17,45 @@
 package capture
 
 import (
+	"fmt"
+
 	"github.com/nmstate/nmpolicy/nmpolicy/internal/resolver"
 	"github.com/nmstate/nmpolicy/nmpolicy/types"
 )
 
 type Result struct {
+	lexer          Lexer
+	parser         Parser
+	resolver       Resolver
 	resolverResult resolver.Result
 }
 
-func (r Result) ResolveCaptureEntryPath(
-	capturePath string) (interface{}, error) {
-	return nil, nil
+func NewResult(l Lexer, p Parser, r Resolver, resolverResult resolver.Result) Result {
+	return Result{
+		lexer:          l,
```

Comment:
Why the capture result should include `lexer`, `parser` and `resolver`? IMO it should include `resolverResult` only.
`ResolveCaptureEntryPath` can create it's own `lexer`, `parser`, `resolver`. They are stateless an there is no reason to carry their instances over the flow.
I would keep the `ResolveCaptureEntryPath` in `CaptureEntry` struct as it was before this PR and won't couple the result of `capture.Resolve` to it.

Reply:
I can create a constructor just for unit test this that will pass lexer/parser/resolver so we can stub them.

About going back to `CaptureEntry` the `resolver.Resolver` is returning a `capturedStates` into forms marshaled and unmarshaled and that has nothing to do with `CaptureEntry` wich is just a "raw" at "capture" field, that's why I have chose the name "Result" as a generic term.

Reply:
Created a pair of construcor at `CaptureEntry` one that is used by production code and that initialize lexer/parser/resolver and other for unit tests that allow to pass the stubs.